lib/membrane_webm/parser/ebml.ex

```diff
@@ -0,0 +1,163 @@
+    <<vint::integer-size(vint_width)-unit(8)>> = vint_bytes
+    vint_data = get_vint_data(vint, vint_width)
+    element_id = Integer.to_string(vint, 16)
+
+    %{
+      vint: %{
+        vint: vint,
+        vint_width: vint_width,
+        vint_data: vint_data,
+        element_id: element_id
+      },
+      rest: rest
+    }
+  end
+
+  @doc """
+  Returns an EBML element's name, type and data
+  """
+  def decode_element(bytes) do
+    {id, bytes} = decode_element_id(bytes)
+    {data_size, bytes} = decode_vint(bytes)
+    name = Schema.element_id_to_name(id)
+    type = Schema.element_type(name)
+
+    # `Segment` is a special element requiring different behaviour of the parser
+    # the parser should only wait for more bytes in case of top-level children elements of `Segment`, not the `Segment` itself
+    if name == :Segment do
+      {:ignore_element_header, bytes}
+    else
+      case split_bytes(bytes, data_size) do
+        {data, bytes} ->
+          # TODO: remove; only for debugging
+          if name == :Unknown do
+            IO.warn("unknown element ID: #{id}")
+          end
+
+          {name, type, data, bytes}
+
+        :need_more_bytes ->
+          :need_more_bytes
+      end
+    end
+  end
+
+  defp split_bytes(bytes, how_many) do
+    if how_many > byte_size(bytes) do
+      :need_more_bytes
+    else
+      <<bytes::binary-size(how_many), rest::binary>> = bytes
+      {bytes, rest}
+    end
+  end
+
+  @doc """
+  Returns the `EBML Element ID` of the given VINT.
+
+  EMBL elements are identified by the hex representation of the entire leading VINT including WIDTH, MARKER and DATA
+  """
+  def decode_element_id(<<first_byte::unsigned-size(8), _rest::binary>> = bytes) do
+    vint_width = get_vint_width(first_byte)
+    <<vint_bytes::binary-size(vint_width), rest::binary>> = bytes
+    <<vint::integer-size(vint_width)-unit(8)>> = vint_bytes
+
+    {Integer.to_string(vint, 16), rest}
+  end
+
+  @doc "Returns the number encoded in the VINT_DATA field of the VINT"
+  def decode_vint(<<first_byte::unsigned-size(8), _rest::binary>> = bytes) do
+    vint_width = get_vint_width(first_byte)
+    <<vint_bytes::binary-size(vint_width), rest::binary>> = bytes
+    <<vint::integer-size(vint_width)-unit(8)>> = vint_bytes
+
+    {get_vint_data(vint, vint_width), rest}
+  end
+
+  # the numbers are bit masks for extracting the data part of a VINT
+  defp get_vint_data(vint, vint_width) do
+    case vint_width do
+      1 -> vint &&& 18_446_744_073_709_551_743
+      2 -> vint &&& 18_446_744_073_709_567_999
+      3 -> vint &&& 18_446_744_073_711_648_767
+      4 -> vint &&& 18_446_744_073_977_987_071
+      5 -> vint &&& 18_446_744_108_069_289_983
+      6 -> vint &&& 18_446_748_471_756_062_719
+      7 -> vint &&& 18_447_307_023_662_972_927
+      8 -> vint &&& 18_518_801_667_747_479_551
+    end
+  end
+
+  defp get_vint_width(byte) do
+    cond do
+      (byte &&& 0b10000000) == 0b10000000 -> 1
```

Comment:
You can use erlang's `:erlang.band(byte, ...) > 0 ->`.  Basically if you do the `and` operation on a number and it matches the pattern here if has to be greater than zero :)